psi4/driver/driver.py

```diff
@@ -872,6 +872,70 @@ def properties(*args, **kwargs):
     else:
         return core.variable('CURRENT ENERGY')
 
+def optimize_geometric(name, **kwargs):
+
+    # will move imports to top of file when i'm sure where this method goes
+    # TODO: check if GeomeTRIC is installed at the start
+    import qcengine as qcng
+    import qcelemental as qcel
+    from qcelemental.models import OptimizationInput
+
+    core.print_out('\n  Optimizing system with GeomeTRIC through QCEngine\n')
+
+    # TODO: return dummy wavefunction? Or run gradient at optimized geometry to get a wfn?
+    return_wfn = kwargs.pop('return_wfn', False)
+    if return_wfn:
+        raise ValidationError(f"return_wfn=True is not supported with GeomeTRIC.")
+
+    # TODO: fully account for basis being in name, and/or composite methods
+    if '/' in name:
+        raise ValidationError(f'Composite methods like {name} are not yet supported with GeomeTRIC')
+                                 
+    # Collect all user-specified keywords
+    keywords = {}
+    for opt in core.get_global_option_list():
+        if core.has_global_option_changed(opt):
+            keywords[opt] = core.get_global_option(opt)
+    basis = keywords['BASIS']
+
+    # Make sure the molecule the user provided is the active one
+    molecule = kwargs.pop('molecule', core.get_active_molecule())
+
+    # If we are freezing cartesian, do not orient or COM
+    if core.get_local_option("OPTKING", "FROZEN_CARTESIAN"):
+        molecule.fix_orientation(True)
+        molecule.fix_com(True)
+    molecule.update_geometry()
+
+    input_data = OptimizationInput(**{
+            'initial_molecule' : molecule.to_schema(dtype=2),
+            'input_specification' : {
+                'model' : { 'method' : name,
+                            'basis' : basis,
+                },
+                'keywords' : keywords,
+            },
+            'keywords' : {'program' : 'psi4'},
+    })
+
+    # TODO: return history if requested
+    ret = qcng.compute_procedure(input_data, "geometric", raise_error=True, local_options={})
+    last_iteration = ret.trajectory[-1]
+    return_energy = last_iteration.properties.return_energy
+
+    # Update molecule with optimized geometry
+    # is there a better way to do this?
+    molecule_opt = last_iteration.molecule # qcelemental.models.molecule.Molecule
```

Comment:
you could pull the `molecule_opt.geometry` and pass the return to `molecule.set_geometry`, I think.